src/Type/MixedType.php

```diff
@@ -169,7 +169,15 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $uni
 
 	public function setExistingOffsetValueType(Type $offsetType, Type $valueType): Type
 	{
-		return new self($this->isExplicitMixed);
+		$types = [
+			new ArrayType(new MixedType(), new MixedType()),
+			new ObjectType(ArrayAccess::class),
+			new NullType(),
+		];
+		if ($offsetType->isInteger()->yes()) {
```

Comment:
I think it should be
```suggestion
		if (!$offsetType->isInteger()->no()) {
```

I set $mixed1[$mixed2] ; $mixed1 could still be a string.

Also if I set $mixed1[$intOrString] ; $mixed1 can stil be a string because maybe $intOrString is never a string finally...